File: Animales.java
import java.util.ArrayList;
import java.util.Scanner;
import java.util.Random;

public class Animales {
    Scanner leer = new Scanner(System.in);
    Random ran = new Random();
    ArrayList<Animales> listaAnimales = new ArrayList<>();
    private String tipoDeAnimal;
    private String fechaDeNacimiento;
    private String fechaLlegada;
    private float peso;
    private String frecuenciaDeAlimentacion;
    private String tiposDeAlimentacion;
    private ArrayList<String> enfermedades;
    private int id;
    private boolean vacunas;

    public Animales(String tipoDeAnimal, int id, String fechaDeNacimiento, String fechaLlegada, float peso, String frecuenciaDeAlimentacion, String tiposDeAlimentacion, boolean vacunas) {
        this.tipoDeAnimal = tipoDeAnimal;
        this.id = id;
        this.fechaDeNacimiento = fechaDeNacimiento;
        this.fechaLlegada = fechaLlegada;
        this.peso = peso;
        this.frecuenciaDeAlimentacion = frecuenciaDeAlimentacion;
        this.tiposDeAlimentacion = tiposDeAlimentacion;
        this.vacunas = vacunas;
        this.enfermedades = new ArrayList<>();
    }
    public Animales() {
        //Este lo use para poder crear el objeto en main sin tener que mandar ningun atributo
    }
    public String getTipoDeAnimal() {
        return tipoDeAnimal;
    }

    public int getId() {
        return id;
    }

    public String getFechaDeNacimiento() {
        return fechaDeNacimiento;
    }

    public String getFechaLlegada() {
        return fechaLlegada;
    }

    public float getPeso() {
        return peso;
    }

    public String getFrecuenciaDeAlimentacion() {
        return frecuenciaDeAlimentacion;
    }

    public String getTiposDeAlimentacion() {
        return tiposDeAlimentacion;
    }

    public boolean getVacunas() {
        return vacunas;
    }

    public ArrayList<String> getEnfermedades() {
        return enfermedades;
    }

    public void setEnfermedades(ArrayList<String> enfermedades) {
        this.enfermedades = enfermedades;
    }


    public void RegistrarAnimal() {
        System.out.print("Ingrese tipo de Animal:");
        String tipoDeAnimal = leer.nextLine();
        int id = ran.nextInt(200) + 1;
        ValidarId(id);
        System.out.print("\nIngrese fecha de nacimiento con el formato Dia/Mes/Año:");
        String fechaDeNacimiento = leer.nextLine();
        System.out.print("\nIngrese Fecha de llegada con el formato Dia/Mes/Año:");
        String fechaLlegada = leer.nextLine();
        System.out.print("\nIngrese peso:");
        float peso = leer.nextFloat();
        leer.nextLine();
        System.out.print("\nIngrese Frecuencia de alimentacion:");
        String frecuenciaDeAlimentacion = leer.nextLine();
        System.out.print("\nIngrese tipos de alimentacion:");
        String tiposDeAlimentacion = leer.nextLine();
        System.out.print("\nTiene vacunas 'Si' No tiene vacunas 'No':");
        String vac = leer.nextLine();
        boolean vacunas = false;
        if(vac.equals("Si")){
            vacunas = true;
        }
        ArrayList<String> enfermedades = new ArrayList<>();
        System.out.print("\nIngrese enfermedades, cuando acabe ingrese Fin:");
        String enfermedad = leer.nextLine();
        while (!enfermedad.equals("Fin")) {
            enfermedades.add(enfermedad);
            System.out.print("Ingrese enfermedad:");
            enfermedad = leer.nextLine();
        }

        Animales animal = new Animales(tipoDeAnimal, id, fechaDeNacimiento, fechaLlegada, peso, frecuenciaDeAlimentacion, tiposDeAlimentacion, vacunas);
        animal.setEnfermedades(enfermedades);
        listaAnimales.add(animal);
    }

    public void MostrarAnimales() {
        for (int i = 0; i < listaAnimales.size(); i++) {
            Animales animal = listaAnimales.get(i);
            System.out.printf("Tipo de animal:%s\nSu id:%d\nFecha de nacimiento:%s\nFecha de llegada:%s\nPeso:%.2fKg\nFrecuencia de alimentacion:%s\nTipos de alimentacion:%s\nVacunas:%b\n", animal.getTipoDeAnimal(), animal.getId(), animal.getFechaDeNacimiento(), animal.getFechaLlegada(), animal.getPeso(), animal.getFrecuenciaDeAlimentacion(), animal.getTiposDeAlimentacion(), animal.getVacunas());

            ArrayList<String> enfermedades = animal.getEnfermedades();
            for (int j = 0; j < enfermedades.size(); j++) {
                System.out.printf("Enfermedad %d: %s\n", j + 1, enfermedades.get(j));

            }
        }
    }

    public void ValidarId(int id) {
        for (int i = 0; i < listaAnimales.size(); i++) {
            if (listaAnimales.get(i).getId() == id) {
                id = ran.nextInt(200) + 1;
                i = 0;
            }
        }
    }
    public void EliminarAnimal() {
        for (int i = 0; i < listaAnimales.size(); i++) {
            System.out.printf("\n%d Animal:%s\n ", i+1,listaAnimales.get(i).getTipoDeAnimal());
        }
        System.out.println("Ingrese el numero del animal a eliminar:");
        int opcion = leer.nextInt();
        listaAnimales.remove(opcion-1);
    }


    public void BuscarAnimalPorID(){
        System.out.print("Ingrese el id del animal:");
        int ID = leer.nextInt();
        System.out.println();
        for (int i = 0; i < listaAnimales.size(); i++) {
            if(ID==listaAnimales.get(i).getId()){

                System.out.printf("Tipo de animal:%s\nSu id:%d\nFecha de nacimiento:%s\nFecha de llegada:%s\nPeso:%.2fKg\nFrecuencia de alimentacion:%s\nTipos de alimentacion:%s\nVacunas:%b\n",  listaAnimales.get(i).getTipoDeAnimal(),  listaAnimales.get(i).getId(),  listaAnimales.get(i).getFechaDeNacimiento(),  listaAnimales.get(i).getFechaLlegada(), listaAnimales.get(i).getPeso(),  listaAnimales.get(i).getFrecuenciaDeAlimentacion(), listaAnimales.get(i).getTiposDeAlimentacion(), listaAnimales.get(i).getVacunas());
                Animales animal = listaAnimales.get(i);
                ArrayList<String> enfermedades = animal.getEnfermedades();
                for (int j = 0; j < enfermedades.size(); j++) {
                    System.out.printf("Enfermedad %d: %s\n", j + 1,enfermedades.get(j));

                }

                }
        }
    }


}

File: Animales.py
import random
from typing import List

class Animales:
    def __init__(self, animal_type: str, id: int, birth_date: str, arrival_date: str, weight: float, feeding_frequency: str, feeding_types: str, vaccinated: bool):
        self.animal_type = animal_type
        self.id = id
        self.birth_date = birth_date
        self.arrival_date = arrival_date
        self.weight = weight
        self.feeding_frequency = feeding_frequency
        self.feeding_types = feeding_types
        self.vaccinated = vaccinated
        self.diseases: List[str] = []

    def set_diseases(self, diseases: List[str]):
        self.diseases = diseases


class AnimalManager:
    def __init__(self):
        self.animals: List[Animales] = []
        self.scanner = input

    def register_animal(self):
        animal_type = input("Ingrese tipo de animal: ")
        id = random.randint(1, 200)
        self.validate_id(id)
        birth_date = input("\n Ingrese fecha de nacimiento dia/mes/año: ")
        arrival_date = input("\nIngrese fecha de llegada dia/mes/año: ")
        weight = float(input("\nIngrese peso: "))
        feeding_frequency = input("\nIngrese fecuencia de alimentacion: ")
        feeding_types = input("\nIngrese tipos de alimentacion: ")
        vaccinated = input("\nVacunado 'Yes' No vacunado 'No': ")
        vaccinated = vaccinated == "Yes"
        diseases = []
        disease = input("\nIngrese enfermedad cuando acabe ingrese 'Fin': ")
        while disease != "Fin":
            diseases.append(disease)
            disease = input("Ingresa enfermedad: ")

        animal = Animales(animal_type, id, birth_date, arrival_date, weight, feeding_frequency, feeding_types, vaccinated)
        animal.set_diseases(diseases)
        self.animals.append(animal)

    def display_animals(self):
        for animal in self.animals:
            print(f"Tipo de animal: {animal.animal_type}")
            print(f"ID: {animal.id}")
            print(f"Fecha de nacimiento: {animal.birth_date}")
            print(f"Fecha de llegada: {animal.arrival_date}")
            print(f"Peso: {animal.weight:.2f} kg")
            print(f"Frecuencia de alimentacion: {animal.feeding_frequency}")
            print(f"Tipo de Alimentacion: {animal.feeding_types}")
            print(f"Vacunado: {animal.vaccinated}")
            for i, disease in enumerate(animal.diseases):
                print(f"Enfermedad {i+1}: {disease}")

    def validate_id(self, id: int):
        for animal in self.animals:
            if animal.id == id:
                id = random.randint(1, 200)
                self.validate_id(id)
                return

    def delete_animal(self):
        for i, animal in enumerate(self.animals):
            print(f"\n{i+1} Animal: {animal.animal_type}")
        option = int(input("Ingrese el numero del animal a eliminar: "))
        self.animals.pop(option-1)

    def search_animal_by_id(self):
        id = int(input("Ingrese el ID del animal a buscar: "))
        print()
        for animal in self.animals:
            if animal.id == id:
                print(f"Tipo de animal: {animal.animal_type}")
                print(f"ID: {animal.id}")
                print(f"Fecha de nacimiento: {animal.birth_date}")
                print(f"Fecha de llegada: {animal.arrival_date}")
                print(f"Peso: {animal.weight:.2f} kg")
                print(f"Frecuencia de alimentacion: {animal.feeding_frequency}")
                print(f"Tipo de Alimentacio: {animal.feeding_types}")
                print(f"Vacunado: {animal.vaccinated}")
                for i, disease in enumerate(animal.diseases):
                    print(f"Enfermedad {i+1}: {disease}")

if __name__ == "__main__":
    animal_manager = AnimalManager()
    animal_manager.register_animal()
    animal_manager.display_animals()
    animal_manager.delete_animal()
    animal_manager.search_animal_by_id()
